db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java

```diff
@@ -7545,8 +7545,7 @@ public TransactionController getTransactionCompile() {
             assert !booting:"booting is expected to be false";
 
             LanguageConnectionContext lcc=getLCC();
-            return lcc.getTransactionCompile();
-
+            return lcc == null ? null : lcc.getTransactionCompile();
```

Comment:
Is this necessary to fix the problem or is it just preventing a similar NPE in the future?

Reply:
good point, will check

Reply:
actually it is needed, `pass in language connection context` is not sufficient. if we remove this, `HdfsImportIT.testPropertiesCacheNPE` fails. However i agree that this is not completely understood. I'm creating a new jira.